redisless/src/server.rs

```diff
@@ -188,6 +188,17 @@ fn run_command_and_get_response<T: Storage>(
                 }
                 None => protocol::NIL.to_vec(),
             },
+            Command::GetSet(k, v) => {
+                let response = match lock_then_release(storage).get(k.as_slice()) {
```

Comment:
You're better to use only once `lock_then_release(storage)` for the whole block. Otherwise, you lock twice instead of once (it's a costly operation), and between the 2 locks, you risk losing the atomicity of that change.